src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id AS car_id, manufacturers.id AS manufacturer_id,"
+                + " manufacturers.name AS manufacturer_name, manufacturers.country"
+                + " AS manufacturer_country, model FROM cars c"
+                + " JOIN manufacturers  ON c.manufacturer_id = manufacturers.id"
+                + " WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+            assert car != null;
+            car.setDrivers(getCarDrivers(id));
+            return Optional.of(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id + ". ", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, manufacturers.id AS manufacturer_id,"
+                + " manufacturers.name AS manufacturer_name,"
+                + " manufacturers.country AS manufacturer_country, model "
+                + "FROM cars c JOIN manufacturers  ON c.manufacturer_id = manufacturers.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement
+                         = connection.prepareStatement(query)) {
+            updateCarStatement.setLong(1, car.getManufacturer().getId());
+            updateCarStatement.setString(2, car.getModel());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+    }
```

Comment:
There, when you update car, you should remove all drivers from it, and then add them all again to update